VirtualReality/qSlicerVirtualRealityModuleWidget.cxx

```diff
@@ -125,23 +125,29 @@ void qSlicerVirtualRealityModuleWidget::updateWidgetFromMRML()
 
   wasBlocked = d->DesiredUpdateRateSlider->blockSignals(true);
   d->DesiredUpdateRateSlider->setValue(vrViewNode != NULL ? vrViewNode->GetDesiredUpdateRate() : 0);
+  this->onDesiredUpdateRateChanged(vrViewNode != NULL ? vrViewNode->GetDesiredUpdateRate() : 0);
```

Comment:
Could you please also change these "NULL"s to "nullptr"s? Once you're there :) Thanks

Reply:
I don't see the point in blocking signals and then calling the on... function manually. What's the reason for this?

Reply:
No real reason. This way does everything in a single pass with no Qt event system triggered. Could do it the other way too. Same effect probably.

Reply:
Either you want the slot to be called or not. If you do, then you don't block the signal, or vice versa. The purpose of the updateWidgetFromMRML is to update the UI elements to reflect the MRML state. There should be no need to call the handler functions that serve the opposite purpose. So: what exactly is it that is needed to be done here, for which you added this call?

Reply:
The slot performs some logic to calculate the magnification value and then update an associated label. I want that to happen as right now the widget is not fully updated from the MRML data (the slider is, but the label is unchanged). Instead of copy-pasting the code, I called the slot as a function instead of a slot.

Reply:
In that case please separate this function out in its own function. There should be no MRML*->Set* call done from updateWidgetFromMRML